### PR TITLE
Adds batch execution matcher

### LINKED ISSUE #107: Add a hamcrest matcher for batch executions

### PR DESCRIPTION
**Pull request by adamdougal**
Also adds ability to retrieve variable types of prepared statements
executed within batches.

Fixes #107.

### CODE REVIEW

File: java-client/src/main/java/org/scassandra/matchers/BatchExecutionMatcher.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
+package org.scassandra.matchers;
```
Comment (chbatey) [inline java-client/src/main/java/org/scassandra/matchers/BatchExecutionMatcher.java]: Needs license

File: java-client/src/main/java/org/scassandra/http/client/BatchExecution.java [hunk 2]
```diff
@@ -78,8 +84,8 @@ public static BatchExecutionBuilder builder() {
 
     public static class BatchExecutionBuilder {
         private List<BatchQuery> batchQueries;
-        private String consistency;
-        private BatchType batchType;
+        private String consistency = "ONE";
```
Comment (chbatey) [inline java-client/src/main/java/org/scassandra/http/client/BatchExecution.java]: Can you java doc the new defaults on the static builder method?

File: java-client/src/main/java/org/scassandra/http/client/BatchQuery.java [hunk 3]
```diff
@@ -15,20 +15,25 @@
  */
 package org.scassandra.http.client;
 
+import org.scassandra.cql.CqlType;
```
Comment (chbatey) [inline java-client/src/main/java/org/scassandra/http/client/BatchQuery.java]: can you put custom imports after java please

File: java-client/src/main/java/org/scassandra/matchers/BatchExecutionMatcher.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import org.scassandra.http.client.BatchExecution;
+import org.scassandra.http.client.BatchQuery;
+
+import java.util.List;
+
+public class BatchExecutionMatcher extends ScassandraMatcher<List<BatchExecution>, BatchExecution> {
+
+    private final BatchExecution expectedBatchExecution;
+
+    public BatchExecutionMatcher(BatchExecution expectedBatchExecution) {
```
Comment (chbatey) [inline java-client/src/main/java/org/scassandra/matchers/BatchExecutionMatcher.java]: given you have a factory method can you make this package private so people don't depend on the constructor? (the others should be too, was a mistake)

File: java-client/src/test/java/org/scassandra/matchers/BatchStatementMatcherTest.java [hunk 5]
```diff
@@ -0,0 +1,801 @@
...
+                .withBatchQueries(BatchQuery.builder().withQuery("some query").build(), unmatchingBatchQuery)
+                .withBatchType(LOGGED)
+                .build();
+
+        BatchExecution expectedExecution = BatchExecution.builder()
+                .withBatchQueries(BatchQuery.builder().withQuery("some query").build())
+                .withBatchType(LOGGED)
+                .build();
+
+        BatchExecutionMatcher underTest = new BatchExecutionMatcher(expectedExecution);
```
Comment (chbatey) [inline java-client/src/test/java/org/scassandra/matchers/BatchStatementMatcherTest.java]: can you use the status method in the tests? it is unused atm

File: server/src/main/scala/org/scassandra/server/actors/BatchHandler.scala [hunk 6]
```diff
@@ -74,7 +74,7 @@ class BatchHandler(activityLog: ActivityLog,
               // Decode query parameters using the prepared statement metadata.
               val dataTypes = prepared.preparedMetadata.columnSpec.map(_.dataType)
               val values = extractQueryVariables(queryText, Some(byteValues), dataTypes).getOrElse(Nil)
-              BatchQuery(queryText, BatchQueryKind.Prepared, values)
```
Comment (chbatey) [inline server/src/main/scala/org/scassandra/server/actors/BatchHandler.scala]: Can we add a test that includes some variables to show they are captured?

File: java-client/src/test/java/org/scassandra/matchers/BatchStatementMatcherTest.java [hunk 7]
```diff
@@ -0,0 +1,801 @@
...
+
+        //when
+        boolean matched = underTest.matchesSafely(newArrayList(actualExecution, unmatchingBatchExecution));
+
+        //then
+        assertTrue(matched);
+    }
+}
```
Comment (chbatey) [inline java-client/src/test/java/org/scassandra/matchers/BatchStatementMatcherTest.java]: new line at end of file

**Comment by chbatey** [comment]
Btw can you raise this from a feature branch rather than your master?

File: java-client/src/main/java/org/scassandra/matchers/BatchExecutionMatcher.java (hunk 1 above)
Comment (adamdougal) [inline java-client/src/main/java/org/scassandra/matchers/BatchExecutionMatcher.java]: Added

File: java-client/src/test/java/org/scassandra/matchers/BatchStatementMatcherTest.java (hunk 5 above)
Comment (adamdougal) [inline java-client/src/test/java/org/scassandra/matchers/BatchStatementMatcherTest.java]: Do you mean static method?

Comment (chbatey) [inline java-client/src/test/java/org/scassandra/matchers/BatchStatementMatcherTest.java]: yep

**Comment by adamdougal** [comment]
Closing in favour of #185

**Pull request closed without merging**